index.js

```diff
@@ -173,14 +196,12 @@ function compression (options) {
         return
       }
 
-      // compression method
-      var accept = accepts(req)
-      var method = accept.encoding(['gzip', 'deflate', 'identity'])
+      // force proper priorization
+      var headers = objectAssign({}, req.headers, { 'accept-encoding': prioritize(req.headers['accept-encoding']) })
 
-      // we really don't prefer deflate
-      if (method === 'deflate' && accept.encoding(['gzip'])) {
-        method = accept.encoding(['gzip', 'identity'])
-      }
+      // compression method
+      var accept = accepts(objectAssign({}, res, { headers: headers }))
```

Comment:
AFAIK the accepts API expects the request object, not a response object clone.

Reply:
Good catch, this is a bug (although it didn't actually effect anything because `accepts` library only reads the headers). This has been fixed.

Reply:
I'm not sure if this is even cloning the `res` correctly, anyhow; it seems to only copy over the own, enumerable properties; all the methods from the prototype and such are lost, for instance.

Reply:
I've updated this code so that we don't pass in the entire (cloned) request and instead with just pass in the cloned headers. Thats all the accepts function reads from anyway.

Reply:
So that of course does end up violating the API contract of that dependency. We should either hard-pin that dependency so that we have to manually bump the version for any changes since this would now be relying on the internal behavior of that dependency to not access anything in outside of that one property, or perhaps get the dependency to change/clarify the contract to only being the headers. If we do go the pin dependency route, we'll want to check that if it then passes the `req` object to a dependency of it's own, that that is pinned within that module as well. This will prevent the possibility of sudden breakage when dependencies, dependencies of dependencies, etc. update